src/ivc/incrementally_verifiable_computation.rs

```diff
@@ -71,19 +81,245 @@ where
+    ///    receive generates [`PlonkInstance`] (`u`) & [`crate::plonk::PlonkWitness`]
+    ///
+    /// 6. **Creating StepCircuitContext**: For both primary and secondary circuits, a
+    ///    [`StepCircuitContext`] is created, encapsulating the step circuit, its relaxed trace, and
+    ///    the initial and final states.
+    ///
+    /// 7. **Finalizing IVC Instance**: The function concludes by constructing and returning the
+    ///    `IVC` instance, which holds the contexts for both circuits.
     pub fn new<const T: usize, RP1, RP2>(
-        _pp: &PublicParams<'_, A1, A2, T, C1, C2, SC1, SC2, RP1, RP2>,
-        _primary: SC1,
-        _z0_primary: [C1::Scalar; A1],
-        _secondary: SC2,
-        _z0_secondary: [C2::Scalar; A2],
+        pp: &PublicParams<'_, A1, A2, T, C1, C2, SC1, SC2, RP1, RP2>,
+        primary: SC1,
+        z0_primary: [C1::Scalar; A1],
+        secondary: SC2,
+        z0_secondary: [C2::Scalar; A2],
     ) -> Result<Self, Error>
     where
         RP1: ROPair<C1::Scalar, Config = MainGateConfig<T>>,
         RP2: ROPair<C2::Scalar, Config = MainGateConfig<T>>,
     {
         // TODO #31
-        todo!("Logic at `RecursiveSNARK::new`")
+        info!("start ivc base case");
+
+        let (primary_config, mut primary_td) =
+            pp.primary.prepare_td(&[C1::Scalar::ZERO, C1::Scalar::ZERO]);
+        debug!("primary step circuit configured");
+
+        let (secondary_config, mut secondary_td) = pp
+            .secondary
+            .prepare_td(&[C2::Scalar::ZERO, C2::Scalar::ZERO]);
+        debug!("secondary step circuit configured");
+
+        let secondary_ps = secondary_td
+            .plonk_structure()
+            .ok_or(Error::MissedPlonkStructure { is_primary: false })?;
+        let primary_cross_term_commits_len =
+            secondary_ps.get_degree_for_folding().saturating_sub(1);
+
+        let primary_ps = primary_td
+            .plonk_structure()
+            .ok_or(Error::MissedPlonkStructure { is_primary: true })?;
+        let secondary_cross_term_commits_len =
+            primary_ps.get_degree_for_folding().saturating_sub(1);
+
+        let primary_public_params_hash = public_params::calc_digest::<C1, C2, C2, RP1, RP2>(
+            pp.primary.params(),
+            &primary_ps,
+            pp.secondary.params(),
+            &secondary_ps,
+        )?;
+        let secondary_public_params_hash = public_params::calc_digest::<C1, C2, C1, RP1, RP2>(
+            pp.primary.params(),
+            &primary_ps,
+            pp.secondary.params(),
+            &secondary_ps,
+        )?;
+
+        debug!("cross term commits len: primary={primary_cross_term_commits_len}, secondary={secondary_cross_term_commits_len}");
+
+        let (primary_ctx, primary_plonk_instance) = {
+            let mut layouter =
+                SingleChipLayouter::<'_, C1::Scalar, _>::new(&mut primary_td, vec![])?;
+
+            let primary_assigned_z0: [_; A1] = layouter.assign_region(
```

Comment:
Instead of assign `z0` here, it is better to treat `z0` as part of `StepInputs`, then create the circuit:
`IVCStepCircuit::new(step_inputs)`.

Reply:
Yeah, I already gave you a branch where it's done that way. Will add it here